jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ConnectorTimeoutTest.java

```diff
@@ -629,17 +628,17 @@ public void run()
                     }
                     if (Content.Chunk.isFailure(chunk))
                     {
+                        bufferAccumulator.release();
                         callback.failed(chunk.getFailure());
                         return;
                     }
                     // copy buffer
-                    bufferAccumulator.copyBuffer(chunk.getByteBuffer().slice());
+                    bufferAccumulator.append(chunk);
                     chunk.release();
                     if (chunk.isLast())
                     {
                         // write accumulated buffers
-                        RetainableByteBuffer buffer = bufferAccumulator.toRetainableByteBuffer();
-                        response.write(true, buffer.getByteBuffer(), Callback.from(buffer::release, callback));
+                        response.write(true, bufferAccumulator.getByteBuffer(), Callback.from(bufferAccumulator::release, callback));
```

Comment:
```suggestion
                        bufferAccumulator.writeTo(response, true, callback);
```

Please check behavior of `DynamicCapacity.writeTo()` in case of multiple buffers.

Reply:
I've replaced `response.write()` with `bufferAccumulator.writeTo()`. But the callback should actually be `Callback.from(bufferAccumulator::release, callback)` as the `writeTo()` contract does not guarantee that resources will be released when it's done: calling `writeTo()` on a simple `RetainableByteBuffer` acquired from a pool won't release the rbb back to the pool.